File: old/p20.rb
class Integer
  def factors_sum()
    sum = 0
  	(2...Math.sqrt(self).floor).each do |i|
  	  if self % i == 0
  	  	sum += i
  	  	sum += self / i
  	  end
  	end
  	sum += 1
  	if Math.sqrt(self).floor == Math.sqrt(self)
  	 	sum += Math.sqrt(self).to_i
  	end
  	sum
  end
end

am_sub = 0

for i in (2...10000)
	for j in (2...10000)
		if j.factors_sum == i && i.factors_sum == j && j != i
			#puts i.to_s + ", " + j.to_s
			am_sub += j
		end
	end
end

puts am_sub

File: old/p20.cpp
#include <iostream>
#include <math.h>

using namespace std;

int factors(int val){
	int sum = 0;
	for (int i = 2; i < floor(sqrt(val)); i++){
		if (val % i == 0){
			sum += i;
			sum += val / i;
		}
	}
	sum++;
	if (floor(sqrt(val)) == sqrt(val)){
		sum += sqrt(val);
	}
	return sum;
}

int main(){
	int am_sub = 0;
	for (int i = 2; i < 10000; i++){
		for (int j = 2; j < 10000; j++){
			if (factors(i) == j && factors(j) == i && j != i){
				am_sub += i;
			}
		}
	}
	cout << am_sub;
	return 0;
}